rpcserver.go

```diff
@@ -7730,6 +7730,34 @@ func (r *rpcServer) AddInvoice(ctx context.Context,
 		maxUnits = assetUnits.ToUint64()
 	}
 
+	// Since we used a different oracle price query above calculate the max
+	// amount of units, we want to add some breathing room to account for
+	// price fluctuations caused by the small time delay, plus the fact that
+	// the agreed upon quote may be different. If we don't add this safety
+	// window the peer may allow a routable amount that evaluates to less
+	// than what we ask for.
+	tolerance := rfqmath.NewBigIntFromUint64(
+		r.cfg.RfqManager.GetPriceDeviationPpm(),
+	)
+
+	// Parse the max asset units to the rfqmath type.
+	maxMathUnits := rfqmath.NewBigIntFromUint64(maxUnits)
+
+	// Calculate the tolerance margin.
+	toleranceUnits := maxMathUnits.Mul(tolerance).Div(
```

Comment:
See `WithinTolerance()`.

Reply:
The advantage of using the existing method is that it's a standardized check elsewhere, and the method has tests (also note the recent bug fix in that area).

Reply:
Ah I think the difference here is that we want the absolute value that would be in tolerance vs determining if something is already within our tolerance range. 

Either way, I think we should extract this to a function so we can write tests for it. Generally we should try to minimize the amount of new biz logic we add to the rpc server functinos.

Reply:
We should also add some logging here so we can pick up on any abnormalities w.r.t the calculation.